catalog/video/video-admin-plugin/src/main/java/org/codice/alliance/video/ui/service/StreamMonitorHelper.java

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.ui.service;
+
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+
+import org.apache.commons.collections.MapUtils;
+import org.codice.alliance.video.stream.mpegts.StreamMonitor;
+import org.codice.alliance.video.stream.mpegts.UdpStreamMonitor;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+
+public class StreamMonitorHelper implements StreamMonitorHelperMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StreamMonitorHelper.class);
+
+    private static final String MONITORING = "monitoring";
+
+    private static final String SERVICE_PID = "service.Pid";
+
+    private static final String START_TIME = "startTime";
+
+    private ObjectName objectName;
+
+    private MBeanServer mBeanServer;
+
+    private BundleContext bundleContext;
```

Comment:
I'm not sure that we want to hold on to a `BundleContext` in an instance variable. I _think_ that we're supposed to get it from `FrameworkUtil` every time we want to use it.